### PR TITLE
[WIP]Add Scala Native support

### PR DESCRIPTION
**Pull request by exoego**

### CODE REVIEW

**Comment by eed3si9n** [comment]
Thank you!
You might need to copy this script into the repo - https://raw.githubusercontent.com/scala-native/scala-native/master/scripts/travis_setup.sh

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,12 @@
-dist: trusty
+dist: xenial
 group: stable
+sudo: required
 
 language: scala
-jdk: oraclejdk8
+jdk: openjdk11
```
Comment (eed3si9n) [inline .travis.yml]: Maybe we should try openjdk8 first.
For xenial I had to do something like https://github.com/typelevel/jawn/blob/e2e77044a3913c457f91b26ebe8c330c116a4714/.travis.yml.
Alternatively you could use SDKMAN - http://eed3si9n.com/all-your-jdks-on-travis-ci-using-sdkman

**Comment by exoego** [comment]
🤔 Hmm, `libunwind` and `no main class` issues are now resolved, but [nativeLink faied due to java.nio.file.ClosedFileSystemException](https://travis-ci.org/sbt/sbt-projectmatrix/builds/532085178#L842).

File: .travis.yml (hunk 1 above)
Comment (exoego) [reply .travis.yml]: Added openjdk8 in https://github.com/sbt/sbt-projectmatrix/pull/12/commits/f749d7d896e8a90fc9b9dab3c0c77f6d2ac6bb92

File: src/sbt-test/projectMatrix/native/build.sbt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+//   .aggregate(core.projectRefs ++ app.projectRefs: _*)
+//   .settings(
+//   )
+
+lazy val core = (projectMatrix in file("core"))
+  .settings(
+    name := "core",
+    mainClass in (Compile, run) := Some("a.CoreMain")
+  )
+  .nativePlatform(scalaVersions = Seq("2.11.12", "2.12.4"))
```
Comment (kevinwright) [inline src/sbt-test/projectMatrix/native/build.sbt]: scala-native doesn't exist for 2.12 yet

File: src/sbt-test/projectMatrix/native/test [hunk 3]
```diff
@@ -0,0 +1,5 @@
+> nativeLink
+
+$ exists app/target/native-2.12/app-out
+$ exists core/target/native-2.12/core-out
```
Comment (kevinwright) [inline src/sbt-test/projectMatrix/native/test]: scala-native doesn't exist for 2.12 yet, and needs a check for `native-2.11/app-out`

**Comment by exoego** [comment]
Hmm, it seems that we cannot build the project for Scala 2.11 with sbt 1.2.x, since `scripted 1.x` is not published for Scala 2.12.

**Comment by exoego** [comment]
`scripted-plugin` droped Scala 2.11 after `1.0.0-M4`, therefore it seems unlikely `scripted-plugin 1.x` is published for Scala 2.11.
So, cross-building for Scala 2.11 and 2.12 is not possible as of today, IIUC.

I think we may revisit Scala Native support when Scala Native support Scala 2.12.

File: .travis.yml [hunk 4]
```diff
@@ -9,6 +9,9 @@ language: scala
 matrix:
   include:
     - jdk: openjdk8
+      scala: 2.11.12
```
Comment (eed3si9n) [inline .travis.yml]: This does not make sense. sbt 1 plugins are always published against Scala 2.12.

**Pull request closed without merging**